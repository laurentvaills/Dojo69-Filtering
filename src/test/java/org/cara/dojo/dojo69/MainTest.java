package org.cara.dojo.dojo69;

import java.util.List;

import org.approvaltests.Approvals;
import org.junit.Test;


public class MainTest {
	
	@Test
	public void approvalTests() throws Exception
	{
		// Given
		String expectedFileFilteredPath = "target/test-classes/config-filtered.properties";
		String[] args = new String[]{"target/test-classes/config.properties", expectedFileFilteredPath, "target/test-classes/filter1.properties", "target/test-classes/filter2.properties"};
		
		// When
		Main.main(args);
		
		// Then	
		List<String> keys = TestUtils.getFilteredKeys(expectedFileFilteredPath);
		Approvals.verifyAll("key", keys);
	}

}
