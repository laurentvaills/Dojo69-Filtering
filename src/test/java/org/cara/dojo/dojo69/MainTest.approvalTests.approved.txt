key[0] = key0=key0
key[1] = key-filter1=sucess: value from filter1
key[2] = key-filter2=success: value from filter2
key[3] = key-filter1And2=success: value from filter1
key[4] = key-noFilter=${key-noFilter}
